onboard/catkin_ws/src/task_planning/scripts/task.py

```diff
@@ -66,8 +72,19 @@ def finish(self):
         """Mark the task as finished"""
         self.finished = True
 
+    def unfinish(self):
+        """Mark task as unfinished"""
+        self.finished = False
+
+    def restart(self):
+        self.unfinish()
+        self._on_task_start()
```

Comment:
I don't like how this calls on_task_start, since if you don't call run immediately after, then technically the logic is broken - the on_task_start was not called right before doing the on_task_run. Instead, you might want to set self.started=False, so that run will call the start function on the next call, right before it tries to run.

Reply:
agreed, good point